src/main/java/com/synopsys/integration/blackduck/bdio2/Bdio2FileUploadService.java

```diff
@@ -85,4 +91,18 @@ private Bdio2UploadResult uploadFiles(List<BdioFileContent> bdioFiles, @Nullable
         return jobExecutor.executeJob(bdio2UploadJob);
     }
 
+    // BlackDuck servers 2023.4.1 and later can use only BDIO header information for 
+    // project names and version names and do not need the REST headers which can have issues
+    // with non-ASCII characters.
+	private boolean useOnlyBdioHeaders() {
+		if (blackDuckApiClient.getBlackDuckVersion() != null) {
```

Comment:
Change the return of getBlackDuckVersion() to return an Optional<BlackDuckVersion> and validate isPresent() here instead.